pkg/server/io/helpers.go

```diff
@@ -92,3 +92,13 @@ type stdioPipes struct {
 	stdout io.ReadCloser
 	stderr io.ReadCloser
 }
+
+func newEmptyPipes() (*stdioPipes, *wgCloser) {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &stdioPipes{}, &wgCloser{
```

Comment:
How come earlier in the PR we still create container stdout and stderr when using a binary logger with `NewDiscardLogger` but we set them nil in this helper?

Reply:
We talked offline, for others: `NewContainerIO` pipes are used for the shim to write to and cri-containerd to read from. By creating nil pipes, we can indicate to the shim that we will be using a binary instead and thus the shim should handle setting up the logging. `createContainerLogger` creates pipes for cri-containerd to log to given input written to the pipes created in `NewContainerIO`. The container's IO will operate by reading from the pipes created in `NewContainerIO` that the shim is expected to have written to and attempting to write to the pipes creates in `createContainerLogger` to log. By using `NewDiscardLogger`s in `createContainerLogger` that means that anything read from the pipes created in `NewContainerIO` will be immediately discarded and cri-containerd will not actually do any logging itself.